### PR TITLE
Fixed Bad Static Application Implementation

### PR DESCRIPTION
**Pull request by AmniX**
I have Removed some Bad Implementation of Application which was written in static way.
According to best practices, Context aware components should not be static else they can cause memory leaks

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/util/FireTvUtils.java [hunk 1]
```diff
@@ -1,10 +1,12 @@
 package org.schabi.newpipe.util;
 
+import android.app.Application;
+
 import org.schabi.newpipe.App;
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/util/FireTvUtils.java]:
```suggestion
```

File: app/src/main/java/org/schabi/newpipe/App.java [hunk 2]
```diff
@@ -91,7 +90,6 @@ public void onCreate() {
         }
         refWatcher = installLeakCanary();
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/App.java]:
```suggestion
```

File: app/src/main/java/org/schabi/newpipe/util/FireTvUtils.java [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
 
 public class FireTvUtils {
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/util/FireTvUtils.java]:
```suggestion
public class FireTvUtils {
```

**Comment by avently** [comment]
> According to best practices, Context aware components should not be static else they can cause memory leaks

It's an app context saved into the same app class. It can't make a memory leak because the reference will be cleared on app exit. The best practices you are talking about is used for activities, fragments, etc. These components shouldn't be saved in a static or dynamic ways as you mentioned.

**Comment by Stypox** [comment]
So what should we do with this pr? @avently @undeadfox @TobiGr

**Comment by AmniX** [comment]
> > According to best practices, Context aware components should not be static else they can cause memory leaks
> 
> It's an app context saved into the same app class. It can't make a memory leak because the reference will be cleared on app exit. The best practices you are talking about is used for activities, fragments, etc. These components shouldn't be saved in a static or dynamic ways as you mentioned.

But It's a Good Practice not to keep any static variable of Context Aware Components.
And Application Parent Class is ContextWrapper and Context Wrapper Parent Class is Context.

So Best Practice makes the Application Health At Best State.
And On Other hand, If you want to talk about Application Context Memory leak, Pass this Application context to a asyncTask and run it for 5 minutes, Now Exit from your app and Now You will observe that you app will stay active in background and won't release the memory until your Thread is terminated. 
Cannot Explain More :)

**Pull request closed without merging**